### PR TITLE
UI: move selectable card to add-on

### PR DESCRIPTION
**Pull request by hellobontempo**
Refactors selectable card to match updated designs as well as makes the component accessible. Per design, when a "type" of thing is selected we will use the selectable card. When the selection is a form and there's a default we will use a radio card

1.  Moves `SelectableCard` to addon 
2. Implements HDS card in the component
3. Unifies border radius across card components
4. Fixes card title wrapping for collapsed screen
5. Deletes `<BoxRadio`

In the demo I am using the keyboard:

engines

### mount engines before + after
Screenshot 2023-10-20 at 9 37 52 AM

### auth methods before + after
Screenshot 2023-10-20 at 9 38 13 AM

### removes tooltip (outdated pattern) for enterprise only (disabled) engines and adds enterprise badge instead
Screenshot 2023-10-20 at 9 44 04 AM
Screenshot 2023-10-20 at 9 43 21 AM

### database overview before

db

### database overview after 
_the font sizes will increase slightly when we increase our base font size to 16px_
db

### CODE REVIEW

File: ui/app/components/get-credentials-card.hbs [hunk 1]
```diff
@@ -3,24 +3,28 @@
   SPDX-License-Identifier: BUSL-1.1
 ~}}
 
-<form {{on "submit" this.transitionToCredential}} class="selectable-card is-rounded no-flex">
```
Comment (hellobontempo) [inline ui/app/components/get-credentials-card.hbs:6]: Just wrapped this component in `<Hds::Card>` to remove the styling classes

File: ui/app/components/get-credentials-card.hbs [hunk 2]
```diff
@@ -3,24 +3,28 @@
...
-    @id="search-input-role"
-    @models={{@models}}
-    @selectLimit="1"
-    @backend={{@backend}}
-    @fallbackComponent="input-search"
-    @onChange={{this.handleInput}}
-    @placeholder={{@placeholder}}
-    data-test-search-roles
-  />
-  <button type="submit" class="button is-secondary" disabled={{not this.role}} data-test-get-credentials>
```
Comment (hellobontempo) [inline ui/app/components/get-credentials-card.hbs:23]: updated this button while I was here

File: ui/app/styles/components/action-block.scss [hunk 3]
```diff
@@ -12,7 +12,7 @@
 }
 
 .action-block {
-  @extend .selectable-card;
+  box-shadow: 0 0 0 1px rgba($grey-dark, 0.3);
```
Comment (hellobontempo) [inline ui/app/styles/components/action-block.scss:15]: This seemed to be the only style inherited from `selectable-card`

File: ui/app/styles/components/known-secondaries-card.scss [hunk 4]
```diff
@@ -3,7 +3,7 @@
  * SPDX-License-Identifier: BUSL-1.1
  */
 
-.selectable-card.secondaries {
```
Comment (hellobontempo) [inline ui/app/styles/components/known-secondaries-card.scss:6]: made all references to selectable-card more specific

File: ui/lib/core/addon/templates/components/replication-summary-card.hbs [hunk 5]
```diff
@@ -3,8 +3,9 @@
   SPDX-License-Identifier: BUSL-1.1
 ~}}
 
-<div
```
Comment (hellobontempo) [inline ui/lib/core/addon/templates/components/replication-summary-card.hbs:6]: Screenshot 2023-10-20 at 10 06 42 AM

File: ui/lib/core/app/components/selectable-card.js
Comment (hellobontempo) [inline ui/lib/core/app/components/selectable-card.js:1]: The whole point of this PR...I definitely scope creeped here 🙈

File: ui/public/eco/transform.svg
Comment (hellobontempo) [inline ui/public/eco/transform.svg:1]: Made this icon black to match other icons (was grey)
Screenshot 2023-10-20 at 11 15 43 AM

File: ui/app/styles/components/selectable-card.scss [hunk 6]
```diff
@@ -4,106 +4,20 @@
...
-    text-decoration: none;
-  }
-
-  .button {
-    &:disabled {
-      border-color: $ui-gray-300;
-    }
-  }
-
-  .card-details {
```
Comment (hellobontempo) [inline ui/app/styles/components/selectable-card.scss:41]: I searched for all of these classes and none of them existed

File: ui/app/styles/helper-classes/flexbox-and-grid.scss [hunk 7]
```diff
@@ -104,6 +104,28 @@
   flex: 50%;
 }
 
+// moving away from !important, fresh flex styles below
```
Comment (hellobontempo) [inline ui/app/styles/helper-classes/flexbox-and-grid.scss:107]: All of our flex classes are kind of specific and I really want to move away from using `!important` so opted to make more general styling here instead of things like `is-flex-between`

File: ui/app/styles/helper-classes/flexbox-and-grid.scss [hunk 8]
```diff
@@ -104,6 +104,28 @@
...
+  &.row-wrap {
+    flex-flow: row wrap;
+  }
+
+  // alignment
+  &.space-between {
+    justify-content: space-between;
+  }
+
+  &.row-gap-8px {
```
Comment (hellobontempo) [inline ui/app/styles/helper-classes/flexbox-and-grid.scss:120]: I can also these to `row-gap-m` and we can move to numerical sizing when we've generally cleaned up the CSS to use that more consistently

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs [hunk 9]
```diff
@@ -32,26 +32,31 @@
   <div class="box is-fullwidth is-shadowless has-tall-padding">
     <div class="selectable-card-container {{if (and (eq this.model.connections 403) (eq this.model.roles 403)) 'one-card'}}">
       {{#if this.model.connectionCapabilities.canList}}
-        <SelectableCard
+        <OverviewCard
```
Comment (kiannaquach) [inline ui/app/templates/vault/cluster/secrets/backend/overview.hbs:35]: Thanks for updating this to use `OverviewCard` instead!

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs [hunk 10]
```diff
@@ -32,26 +32,31 @@
...
+          data-test-selectable-card="Connections"
```
Comment (kiannaquach) [inline ui/app/templates/vault/cluster/secrets/backend/overview.hbs:36]: Do we want to update the `data-test` to be a different name since these aren't selectable cards anymore?

File: ui/app/components/get-credentials-card.hbs (hunk 1 above)
Comment (hellobontempo) [reply ui/app/components/get-credentials-card.hbs:6]: before/after
Screenshot 2023-10-20 at 11 50 29 AM

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs [hunk 11]
```diff
@@ -32,26 +32,31 @@
...
-          {{! TODO: Messaging needs massaging }}
-          @subText="The total number of roles that have been set up that you can list."
+          @subText="The total number of roles configured that you have permissions to list."
           @actionText="Create new"
           @actionTo="vault.cluster.secrets.backend.create-root"
-          @queryParam={{"role"}}
-        />
+          @actionQuery={{hash itemType="role"}}
+        >
+          <Hds::Text::Display class="has-top-padding-m" @tag="h2" @size="500">
```
Comment (kiannaquach) [inline ui/app/templates/vault/cluster/secrets/backend/overview.hbs:56]: I've never used this component before! Is this `HDS::Text::Display` component typically used for text with heading tags and `HDS::Text::Code` for code blocks?

File: ui/app/components/mount-backend/type-form.hbs
Comment (hellobontempo) [inline ui/app/components/mount-backend/type-form.hbs:1]: This is the main file I'd like feedback on!

File: ui/app/styles/components/selectable-card.scss [hunk 12]
```diff
@@ -4,106 +4,20 @@
...
-
-      @include until($mobile) {
-        overflow: hidden;
-      }
-    }
+  &:hover:not(.disabled),
+  &:focus,
+  &:focus-visible {
+    cursor: pointer;
+    box-shadow: var(--token-focus-ring-action-box-shadow);
```
Comment (hellobontempo) [inline ui/app/styles/components/selectable-card.scss:11]: https://hds-website-hashicorp.vercel.app/foundations/focus-ring

File: ui/lib/core/addon/components/icon.js [hunk 13]
```diff
@@ -43,6 +43,6 @@ export default class Icon extends Component {
   }
 
   get hsIconClass() {
-    return this.size === '24' ? 'hs-icon-xl' : 'hs-icon-l';
+    return this.size === '24' ? 'hs-icon-xlm' : 'hs-icon-l';
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/icon.js:46]: these icons were a rendering a few pixels too large
```js
// icons.scss
.hs-icon-xlm {
  width: 24px;
  height: 24px;
}

.hs-icon-xl {
  width: 28px;
  height: 28px;
}
```

File: ui/lib/core/addon/components/overview-card.hbs [hunk 14]
```diff
@@ -6,22 +6,21 @@
...
+    <h3 class="has-text-weight-bold is-size-5">{{@cardTitle}}</h3>
     {{#if @actionText}}
-      <LinkTo
-        class="has-icon-right is-ghost is-no-underline has-text-weight-semibold"
+      <Hds::Link::Standalone
+        @icon={{or @actionIcon "chevron-right"}}
+        @iconPosition="trailing"
+        @text={{@actionText}}
         @route={{@actionTo}}
-        @query={{hash @actionQuery}}
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/overview-card.hbs:19]: updated so hash object is passed directly since we can't pass args like 
```js
@actionQuery={{someParam="item"}}

File: ui/app/styles/helper-classes/flexbox-and-grid.scss (hunk 7 above)
Comment (kiannaquach) [reply ui/app/styles/helper-classes/flexbox-and-grid.scss:107]: +1 on moving away from `!important` and love the general styling!

File: ui/lib/core/addon/components/overview-card.hbs [hunk 15]
```diff
@@ -6,22 +6,21 @@
...
   ...attributes
 >
-  <div class="is-flex-between" data-test-overview-card={{@cardTitle}}>
-    <h3 class="title is-5">{{@cardTitle}}</h3>
+  <div class="flex row-wrap space-between has-bottom-margin-m" data-test-overview-card={{@cardTitle}}>
+    <h3 class="has-text-weight-bold is-size-5">{{@cardTitle}}</h3>
     {{#if @actionText}}
-      <LinkTo
-        class="has-icon-right is-ghost is-no-underline has-text-weight-semibold"
+      <Hds::Link::Standalone
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/overview-card.hbs:16]: updated to HDS link which fails gracefully if there is no `query` param. Otherwise, `<LinkTo>` errors when `@actionQuery` is undefined because it demands an object to be passed

File: ui/app/styles/helper-classes/flexbox-and-grid.scss (hunk 8 above)
Comment (kiannaquach) [reply ui/app/styles/helper-classes/flexbox-and-grid.scss:120]: If we're changing to numerical styles, it might be good to with `.row-gap-8`. I noticed in most [bootstrap helper classes](https://getbootstrap.com/docs/4.0/utilities/spacing/#examples) that it's formatted with the number at the end.

File: ui/app/styles/helper-classes/flexbox-and-grid.scss [hunk 16]
```diff
@@ -104,6 +104,28 @@
...
+    row-gap: $spacing-xs;
+  }
+
+  &.column-gap-16px {
```
Comment (kiannaquach) [inline ui/app/styles/helper-classes/flexbox-and-grid.scss:124]:
```suggestion
  &.column-gap-16 {
```

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs (hunk 9 above)
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/secrets/backend/overview.hbs:35]: I know we desperately need a component library but hopefully this cleans things up for our future selves!

File: ui/app/styles/helper-classes/general.scss [hunk 17]
```diff
@@ -114,10 +114,6 @@
 }
 
 // border-radius
-.border-radius-2 {
```
Comment (kiannaquach) [inline ui/app/styles/helper-classes/general.scss:117]: 💖

File: ui/lib/core/addon/components/overview-card.hbs [hunk 18]
```diff
@@ -6,22 +6,21 @@
 <Hds::Card::Container
   @level="mid"
   @hasBorder={{true}}
-  class="overview-card border-radius-2"
```
Comment (kiannaquach) [inline ui/lib/core/addon/components/overview-card.hbs:9]: Thanks for asking this in the hds design channel and removing them! :)

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs (hunk 10 above)
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/secrets/backend/overview.hbs:36]: I can do that

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs (hunk 11 above)
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/secrets/backend/overview.hbs:56]: That was my understanding. There's also `<Hds::Text::Body>` I followed this to determine which component and tags to use https://hds-website-hashicorp.vercel.app/foundations/typography?tab=code#font-styles

File: ui/app/components/mount-backend/type-form.hbs [hunk 19]
```diff
@@ -4,39 +4,30 @@
...
+        data-test-mount-type={{type.type}}
+      >
+        <Icon @name={{or type.glyph type.type}} @size="24" class="has-bottom-margin-xs" />
+        <Hds::Text::Body @tag="h3" @size="300">
+          {{type.displayName}}
+        </Hds::Text::Body>
+        {{#if (and type.requiredFeature (not (has-feature type.requiredFeature)))}}
+          <Hds::Badge @text="Enterprise" @icon="enterprise" @size="small" />
+        {{/if}}
+      </SelectableCard>
```
Comment (kiannaquach) [inline ui/app/components/mount-backend/type-form.hbs:26]: The new `SelectableCard` component is awesome! I may have missed these conversations since I was out last week, but curious why we opted to do `SelectableCard`s instead of `RadioCard`s.

Comment (kiannaquach) [reply ui/app/components/mount-backend/type-form.hbs:26]: I think it was mentioned it was because of ui unification - is HCP doing something similar?

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs [hunk 20]
```diff
@@ -32,26 +32,31 @@
...
-          @total={{if (eq this.model.connections 404) 0 this.model.connections.length}}
           @subText="The total number of connections to external databases that you have access to."
           @actionText="Configure new"
           @actionTo="vault.cluster.secrets.backend.create-root"
-          @queryParam={{"connection"}}
-          @type="role"
-        />
+          @actionQuery={{hash itemType="connection"}}
+        >
+          <Hds::Text::Display class="has-top-padding-m" @tag="h2" @size="500">
```
Comment (kiannaquach) [inline ui/app/templates/vault/cluster/secrets/backend/overview.hbs:42]: Ahh thanks so much for updating this - love this so much better than using the `title-number` class! 💖

File: ui/app/components/mount-backend/type-form.hbs (hunk 19 above)
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs:26]: It was a design decision

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs (hunk 11 above)
Comment (kiannaquach) [reply ui/app/templates/vault/cluster/secrets/backend/overview.hbs:55]: Good to know - so any text we place in the tags should use this `Hds::Text` component?

File: ui/app/components/mount-backend/type-form.hbs [hunk 21]
```diff
@@ -4,39 +4,30 @@
 ~}}
 
 {{#each (array "generic" "cloud" "infra") as |category|}}
-  <h3 class="title box-radio-header">
```
Comment (kiannaquach) [inline ui/app/components/mount-backend/type-form.hbs:7]: I think we can remove `box-radio-header` in the `box-radio.scss` file since `type-form` is the only place that used this class.

File: ui/app/components/mount-backend/type-form.hbs [hunk 22]
```diff
@@ -4,39 +4,30 @@
...
+  <Hds::Text::Display class="has-top-padding-m has-bottom-padding-s" @tag="h2" size="400">
     {{capitalize category}}
-  </h3>
-  <div class="box-radio-container">
```
Comment (kiannaquach) [inline ui/app/components/mount-backend/type-form.hbs:10]: Same here, I think we can delete the `box-radio-container` class from `box-radio.scss`

File: ui/app/components/mount-backend/type-form.hbs [hunk 23]
```diff
@@ -4,39 +4,30 @@
...
-          (or (eq type.type "transform") (eq type.type "kmip") (eq type.type "keymgmt"))
-          (concat
-            type.displayName
-            " is part of the Advanced Data Protection module, which is not included in your enterprise license."
-          )
-          "This secret engine is not included in your license."
-        }}
-      />
+        data-test-mount-type={{type.type}}
+      >
```
Comment (kiannaquach) [inline ui/app/components/mount-backend/type-form.hbs:18]: Do we need to add a tooltipMessage for cards that are `transform`, `kmip`, or `keymgmt`?

File: ui/app/components/mount-backend/type-form.hbs (hunk 19 above)
Comment (kiannaquach) [reply ui/app/components/mount-backend/type-form.hbs:26]: Awesome that makes sense to me!

File: ui/app/templates/vault/cluster/secrets/backend/overview.hbs (hunk 11 above)
Comment (hellobontempo) [reply ui/app/templates/vault/cluster/secrets/backend/overview.hbs:55]: Not necessarily - we just didn't have an equivalent typography helper class so I used this as a stand-in

File: ui/app/components/mount-backend/type-form.hbs (hunk 22 above)
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs:10]: Yep - just haven't pushed the commit where I delete all of Box radio related stuff yet :)

File: ui/app/components/mount-backend/type-form.hbs (hunk 23 above)
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs:18]: Nope - showing a tooltip message when elements are disabled is not accessible and a deprecated pattern

File: ui/app/components/mount-backend/type-form.hbs (hunk 21 above)
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs:7]: I'll be deleting all of the box radio styling + component at the end (for diff purposes waited so the review was easier with fewer files 😄 )

File: ui/lib/core/addon/components/icon.js (hunk 13 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/icon.js:46]: Yeah - I think something may have changed? Or at least I noticed in the mount backend form that all of the structure icons were slightly too large. 

There are still some missing from the HDS library so it'd be good to get helios to add whatever doesn't currently exist so we can replace our old icons.

File: ui/app/components/mount-backend/type-form.hbs [hunk 24]
```diff
@@ -4,39 +4,30 @@
...
+        @onClick={{fn @setMountType type.type}}
         @disabled={{if type.requiredFeature (not (has-feature type.requiredFeature)) false}}
-        @tooltipMessage={{if
-          (or (eq type.type "transform") (eq type.type "kmip") (eq type.type "keymgmt"))
-          (concat
-            type.displayName
-            " is part of the Advanced Data Protection module, which is not included in your enterprise license."
-          )
-          "This secret engine is not included in your license."
-        }}
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs:27]: Yep - consulted Yankun and decided to remove because this is an outdated and non-accessible pattern. We added the `Enterprise` badge instead for these backends (details in PR description above)

File: ui/app/components/mount-backend/type-form.hbs [hunk 25]
```diff
@@ -4,39 +4,30 @@
...
 <div class="field is-grouped box is-fullwidth is-bottomless">
-  <button
-    data-test-mount-next
-    type="button"
-    class="button is-primary"
-    {{on "click" (fn @setMountType this.selection)}}
-    disabled={{not this.selection}}
-  >
-    Next
-  </button>
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs:41]: Yeah - I think the concern was accessibility? But this came update came after consulting design. All the types are keyboard accessible and selectable

[the remark this answers is not in the document]
Comment (zofskeez) [reply ui/app/components/mount-backend/type-form.hbs:41]: Ya I suppose it would be way easier with the keyboard this way, otherwise you are tabbing to get to the _Next_ button a bunch

File: ui/app/components/mount-backend/type-form.hbs [hunk 26]
```diff
@@ -4,39 +4,30 @@
 ~}}
 
 {{#each (array "generic" "cloud" "infra") as |category|}}
-  <h3 class="title box-radio-header">
+  <Hds::Text::Display class="has-top-padding-m has-bottom-padding-s" @tag="h2" size="400">
     {{capitalize category}}
-  </h3>
-  <div class="box-radio-container">
+  </Hds::Text::Display>
+  <div class="flex row-wrap row-gap-8px column-gap-16px has-bottom-padding-m">
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/components/mount-backend/type-form.hbs]: Thanks for catching! Missed this commit when I renamed the class [update class name in template file](https://github.com/hashicorp/vault/pull/23739/commits/2330b3ac61771fe69f052214816c876d41fb90b9)

File: ui/app/styles/helper-classes/general.scss [hunk 27]
```diff
@@ -114,10 +114,6 @@
...
-  border-radius: $radius;
-}
-
 .border-radius-4 {
   border-radius: $radius-large;
 }
```
[the remark this answers is not in the document]
Comment (hellobontempo) [reply ui/app/styles/helper-classes/general.scss:119]: Yeah - I'm going to address sizing variable problems when I update the base font size. I can probably sneak that in there - this is for the code snippet block so I'll confirm if design wants to keep this border radius different on purpose

**Review by zofskeez — APPROVED** [review]
👏
